CHANGELOG.md

```diff
@@ -10,10 +10,77 @@ Changes are grouped as follows
 - `Added` for new features.
 - `Changed` for changes in existing functionality.
 - `Deprecated` for soon-to-be removed features.
+- `Improved` for transparent changes, e.g. better performance.
 - `Removed` for now removed features.
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [5.0.0] - 28-09-22
+### Improved
+- Greatly increased speed of datapoints fetching, especially when asking for...:
+  - Large number of time series (~80+)
+  - Very few time series (1-3)
+  - Any query using a finite `limit`
+  - Any query for `string` datapoints
+- Peak memory consumption is 25-30 % lower when using the new `retrieve_arrays` method (with the same number of `max_workers`).
+- Converting fetched datapoints to a Pandas `DataFrame` via `to_pandas()` (or time saved by using `retrieve_dataframe` directly) has changed from `O(N)` to `O(1)`, i.e. speedup depends on the size and is typically 4-5 orders of magnitude faster (!) (only applies to `DatapointsArray` and `DatapointsArrayList` as returned by the `retrieve_arrays` method).
+- Individual customization of queries is now available for all retrieve endpoints. Previously only `aggregates` could be customized. Now all parameters can be passed either as top-level or as individual settings. This is now aligned with the API.
+- Documentation for the retrieve endpoints has been overhauled with lots of new usage patterns and (better!) examples, check it out!
+- Vastly better test coverage for datapoints fetching logic. You can have increased trust in the results from the SDK!
+
+### Added
+- New optional dependency, `numpy`.
+- A new datapoints fetching method, `retrieve_arrays`, that loads data directly into NumPy arrays for improved speed and lower memory usage.
+- These arrays are stored in the new resource types `DatapointsArray` and `DatapointsArrayList` which offer more efficient memory usage and zero-overhead pandas-conversion.
+
+### Changed
+- The default value for `max_workers`, controlling the max number of concurrent threads, has been increased from 10 to 20.
+- The main way to interact with the `DatapointsAPI` has been moved from `client.datapoints` to `client.time_series.data` to align and unify with the `SequenceAPI`. All example code has been updated to reflect this change. Note however that the `client.datapoints` will still work until the next major release, but will until then issue a `DeprecationWarning`.
```

Comment:
```suggestion
- The main way to interact with the `DatapointsAPI` has been moved from `client.datapoints` to `client.time_series.data` to align and unify with the `SequenceAPI`. All example code has been updated to reflect this change. Note, however, that the `client.datapoints` will still work until the next major release, but will until then issue a `DeprecationWarning`.
```
Alternatively, you could write "NB: The `client.datapoints` will still work until the next..."